d3b_utils/aws_bucket_contents.py

```diff
@@ -0,0 +1,61 @@
+import boto3
+import re
+
+
+def fetch_aws_bucket_obj_info(
+    bucket_name, search_prefixes=None, drop_folders=False, drop_path=False
+):
+    """
+    List the objects of a bucket and the size of each object. Returns a
+    list of dicts.
+
+    :param bucket_name: The name of the bucket.
+    :type bucket_name: str
+    :param search_prefixes: path to the specific folder in the given
+        bucket where the files you want to list are. Automatically drops
+        leading '/'. Include a '/' at the end of folder names. Should look
+        like 'path/to/files/'. This can also be any iteratable.
```

Comment:
Not _any_ iterable, only an iterable of things matching the preceding description.

Reply:
good point. will fix.